charts/redpanda/helpers.go

```diff
@@ -62,8 +62,46 @@ func FullLabels(dot *helmette.Dot) map[string]string {
 		"app.kubernetes.io/component":  Name(dot),
 	}
 
-	// TODO Shouldn't our defaults take precedence over user provided labels?
-	return helmette.Merge(defaults, labels)
+	return helmette.Merge(labels, defaults)
+}
+
+func StatefulSetPodLabelsSelector(dot *helmette.Dot, existingStatefulSetLabelSelector map[string]string) map[string]string {
+	if existingStatefulSetLabelSelector != nil {
+		return existingStatefulSetLabelSelector
+	}
+
+	values := helmette.Unwrap[Values](dot.Values)
+
+	commonLabels := map[string]string{}
+	if values.CommonLabels != nil {
+		commonLabels = values.CommonLabels
+	}
+
+	component := fmt.Sprintf("%s-statefulset",
```

Comment:
nit: use the string helper function please

Reply:
If you would like to use https://github.com/redpanda-data/helm-charts/blob/a0b1456c2274dcf85ea70f10af9b4e1a86846c1c/charts/redpanda/helpers.go#L174-L176 then it will not work as I need to trim not by 63 characters, but 51. 

Maybe you have different function in mind? If labels value can have more than 63, then I can use `cleanForK8s` function. Still it will be breaking change as current implementation uses 51 trim numer.

https://github.com/redpanda-data/helm-charts/blob/a0b1456c2274dcf85ea70f10af9b4e1a86846c1c/charts/redpanda/templates/_statefulset.tpl#L30

Reply:
Aaaaaaaaaaaaaah, My bad I missed the number.